src/transformStateWithClones.js

```diff
@@ -6,8 +6,45 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const allStateVersions = [];
+  let newState = { ...state };
+
+  for (const action in actions) {
+    const type = actions[action].type;
+
+    switch (type) {
+      case 'addProperties': {
+        const addProp = actions[action].extraData;
+
+        for (const prop in addProp) {
+          newState[prop] = addProp[prop];
+        };
+
+        allStateVersions.push({ ...newState });
```

Comment:
Btw, you can push to array after the switch/case so you wouldn't repeat yourself